packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -5330,6 +5330,216 @@ describe('SnapController', () => {
       snapController.destroy();
     });
 
+    it('grants the `endowment:caip25` permission to a Snap with `endowment:ethereum-provider` if the `useCaip25Permission` feature flag is enabled', async () => {
```

Comment:
Can we add some tests for updating a Snap? That it also grants the permission and probably revokes the permission as well?